src/init.jl

```diff
@@ -1,5 +1,13 @@
 value(x) = x
-promote_tspan(u0, p, tspan, prob, kwargs) = tspan
+promote_tspan(u0, p, tspan, prob, kwargs) = _promote_tspan(tspan, kwargs)
+function _promote_tspan(tspan, kwargs)
+    if haskey(kwargs, :dt)
+        tspan1, tspan2, _ = promote(tspan..., kwargs[:dt])
```

Comment:
Probably not worth it since it operates on `Pairs`/`NamedTuple` anyway but one could avoid one lookup:
```suggestion
    if (dt = get(kwargs, :dt, nothing)) !== nothing
        tspan1, tspan2, _ = promote(tspan..., dt)
```